src/main/java/io/cryostat/core/net/JFRConnection.java

```diff
@@ -148,6 +162,86 @@ public synchronized int getPort() {
         }
     }
 
+    public synchronized String getJvmId()
+            throws AttributeNotFoundException, InstanceNotFoundException, MBeanException,
+                    ReflectionException, IOException {
+        if (!isConnected()) {
+            connect();
+        }
+        List<String> attrNames =
+                new ArrayList<>(
+                        Arrays.asList(
+                                "ClassPath",
+                                "Name",
+                                "InputArguments",
+                                "LibraryPath",
+                                "VmVendor",
+                                "VmVersion",
+                                "StartTime"));
+
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(512);
+                DataOutputStream dos = new DataOutputStream(baos)) {
+            for (String attr : attrNames) {
+                Object attrObject =
+                        this.rjmxConnection.getAttributeValue(
+                                new MRI(Type.ATTRIBUTE, ConnectionToolkit.RUNTIME_BEAN_NAME, attr));
+                if (attrObject.getClass().isArray()) {
+                    if (attrObject.getClass().getComponentType().isPrimitive()) {
+                        attrObject = castPrimitiveToObject(attrObject);
+                    }
+                    dos.writeUTF(Arrays.toString((Object[]) attrObject));
```

Comment:
Do we really need to check and determine that the object is actually an array of primitives, then convert that to an array of boxed objects, just to stringify that? I think we can skip the `ArrayUtils.toObject()` calls if the `castPrimitiveToObject` is just refactored to something like `primitiveArrayToString` and calls/returns `Arrays.toString()` instead of `ArrayUtils.toObject()`. Does that make sense?

Reply:
Yeah it makes sense, sorry lots of brainfarts, I will fix it in the morning.